### PR TITLE
Add NativeAOT libraries tests annotations

### PR DESCRIPTION
**Pull request by MichalStrehovsky**
Adds `ActiveIssue`/`Conditional` attributes to libraries tests so that we can get a clean run with NativeAOT.

Starting with System.Collections, System.Runtime, and System.IO.FileSystem.

### CODE REVIEW

File: src/libraries/System.Runtime/tests/System/AttributeTests.cs [hunk 1]
```diff
@@ -154,6 +154,8 @@ class ChildAttributeWithField : ParentAttribute
             public int Field = 0;
         }
 
+
```
Comment (stephentoub) [inline src/libraries/System.Runtime/tests/System/AttributeTests.cs:157]: Nit:
```suggestion
```

File: src/libraries/System.Runtime/tests/System/Runtime/CompilerServices/RuntimeFeatureTests.cs [hunk 2]
```diff
@@ -33,8 +33,16 @@ public static void DynamicCode()
...
+            if (PlatformDetection.IsNativeAot)
+            {
+                Assert.False(RuntimeFeature.IsDynamicCodeSupported);
+                Assert.False(RuntimeFeature.IsDynamicCodeCompiled);
+            }
+            else
+            {
+                Assert.True(RuntimeFeature.IsDynamicCodeSupported);
+                Assert.True(RuntimeFeature.IsDynamicCodeCompiled);
+            }
```
Comment (stephentoub) [inline src/libraries/System.Runtime/tests/System/Runtime/CompilerServices/RuntimeFeatureTests.cs:45]: Nit: could also be:
```suggestion
            Assert.Equal(!PlatformDetection.IsNativeAot, RuntimeFeature.IsDynamicCodeSupported);
            Assert.Equal(!PlatformDetection.IsNativeAot, RuntimeFeature.IsDynamicCodeCompiled);
```

File: src/libraries/System.Runtime/tests/System/Runtime/CompilerServices/RuntimeHelpersTests.cs [hunk 3]
```diff
@@ -145,13 +152,17 @@ public static void PrepareGenericMethod()
                     .MakeGenericMethod(new Type[] { typeof(TestStruct) }).MethodHandle,
                 null);
 
-            // Generic definition without instantiation is invalid
-            Assert.Throws<ArgumentException>(() => RuntimeHelpers.PrepareMethod(typeof(Array).GetMethod("Resize").MethodHandle,
-                null));
+            if (RuntimeFeature.IsDynamicCodeSupported)
+            {
+
```
Comment (stephentoub) [inline src/libraries/System.Runtime/tests/System/Runtime/CompilerServices/RuntimeHelpersTests.cs:157]: Nit:
```suggestion
```

**Review by stephentoub — APPROVED** [review]

File: src/libraries/System.Runtime/tests/System/Runtime/CompilerServices/RuntimeFeatureTests.cs (hunk 2 above)
Comment (MichalStrehovsky) [reply src/libraries/System.Runtime/tests/System/Runtime/CompilerServices/RuntimeFeatureTests.cs:45]: That feels a bit less readable.

This test is currently `SkipOnMono`, but it doesn't have to be (and maybe it shouldn't) - it just needs a way to express the right conditions for Mono and once we do that, the existing structure might be better.

Comment (kant2002) [reply src/libraries/System.Runtime/tests/System/Runtime/CompilerServices/RuntimeFeatureTests.cs:45]: Even if I in same camp as you on preferring `if` for readability (actually I would call suggested snnippet harder to parse). Maybe this way would be slightly better. Just want to check on other people if I'm right what's easier.
```
Assert.Equal(PlatformDetection.IsNativeAot, !RuntimeFeature.IsDynamicCodeSupported);
Assert.Equal(PlatformDetection.IsNativeAot, !RuntimeFeature.IsDynamicCodeCompiled);
```

Comment (MichalStrehovsky) [reply src/libraries/System.Runtime/tests/System/Runtime/CompilerServices/RuntimeFeatureTests.cs:45]: That variation gives me the same pause when reading it

Comment (kant2002) [reply src/libraries/System.Runtime/tests/System/Runtime/CompilerServices/RuntimeFeatureTests.cs:45]: Interesting. Based on my knowledge people with domain knowledge process separate subdomains as completely separate entities with separate rules. That’s similar to how you use `if`. Let’s see what’s Stephen’s opinion.

**Pull request merged**